code-challenges.test.js

```diff
@@ -13,40 +13,214 @@
+    return string.split("").map((value)=>{
+        return codedChars[value] || value }).join("")
+}
+// ----- PASSING TEST CASES-------
+// PASS  ./code-challenges.test.js
+// codedMessage
+//   ✓ Returns given string in coded formart (5 ms)
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
 // --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
+describe("arrayOfGivenLetter", () => {
+        const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+        const letterA = "a"
+        // Expected output: ["Mango", "Apricot", "Peach"]
+        const letterE = "e"
+        // Expected output: ["Cherry", "Blueberry", "Peach"]
+    it("Return only values in array that contain given letter", () => {
+        expect(arrayOfGivenLetter(fruitArray,letterA)).toEqual(["Mango", "Apricot", "Peach"])
+        expect(arrayOfGivenLetter(fruitArray,letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+    })
+})
+//-------Test Case failing due to function not be defined------
+// FAIL  ./code-challenges.test.js
+// arrayOfGivenLetter
+//   ✕ Return only values in array that contain given letter (3 ms)
+// ● arrayOfGivenLetter › Return only values in array that contain given letter
+//   ReferenceError: arrayOfGivenLetter is not defined
 
-const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
-
-const letterA = "a"
-// Expected output: ["Mango", "Apricot", "Peach"]
-const letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
 
 // b) Create the function that makes the test pass.
-
+//------------Pseudo Code-------------
+//Input: array & letter (string)
+//Output: Filtered Array
+//Create a function called arrayOfGivenletter that takes in an array and a single letter(string)
+// You will need to map through your given array to be able to access each word indiviually.
+// then you will split each word that you map through and filter that word.
+// with in your higher order function .filter you will only return the words that the contains the given letter.
+// when comparing the value to the latter we will be using toLowerCase().includes to make sure we dont leave out an matching letters that were upper case.
+// we will then join the returned words back together into the original array.
+// that we filtered the mapped array we will have empty strings return because map will always return the same size array.
+// so we will finally .filter through the array and only return the values that are not emprty strings.
+const arrayOfGivenLetter = (array, letter) => {
+    return array.map((word)=> word.split(' ')
+    .filter((value)=> value.toLowerCase().includes(letter)).join(' '))
+    .filter((value)=> value !='')
+}
+
```

Comment:
Awesome approach! Now that you got the test to pass, consider refactoring. I would look at includes and look at do I need the split and join methods?